_user/voxeldemo/main.c

```diff
@@ -0,0 +1,371 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Self Contained Voxel Space Demo
+ *
+ * Example of user application
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <math.h>
+#include <graph.h>
+
+
+typedef struct _camera_t {
+	float x, y, h;
+	float angle;
+	int horiz;
+	int dist;
+} camera_t;
+
+
+typedef struct _voxel_t {
+	int *bufBack;
+	uint8_t *mapHeight;
+	uint8_t *mapColor;
+	uint32_t *pixels, *palette;
+	unsigned int width;
+	unsigned int height;
+	camera_t cam;
+} voxel_t;
+
+
+static inline int clamp(int x)
+{
+	return x < 0 ? 0 : x > 0xff ? 0xff : x;
+}
+
+
+static int interpolate(uint32_t *tab, uint32_t a, uint32_t b, int n)
+{
+	unsigned int j;
+
+	for (j = 0; j < n; j++) {
+		*tab = ((j * (((uint8_t)(b >> 16))) + (n - j) * (((uint8_t)(a >> 16)))) / n) << 16;
+		*tab |= ((j * (((uint8_t)(b >> 8))) + (n - j) * (((uint8_t)(a >> 8)))) / n) << 8;
+		*tab |= (j * (((uint8_t)(b))) + (n - j) * (((uint8_t)(a)))) / n;
+		tab++;
+	}
+
+	return n;
+}
+
+
+void voxel_free(voxel_t *v)
+{
+	free(v->bufBack);
+	free(v->mapColor);
+	free(v->mapHeight);
+}
```

Comment:
```suggestion
	free(v->mapHeight);
	free(v->palette);
}
```

Reply:
Yes, `palette` is not freed, this is to be fixed.